R/ppc-discrete.R

```diff
@@ -266,6 +279,57 @@ ppc_rootogram <- function(y,
   }
   tyrep <- do.call(rbind, tyrep)
   tyrep[is.na(tyrep)] <- 0
+
+  #Discrete style
```

Comment:
@jgabry this is at the same time a call for your opinion.
Now the function is housing two very different functions inside it, the traditional three rootograms, and the discrete one. The discrete one has a separate logical branch that has its own return statement.

They differ in
1. Data generation (slightly) mean vs. median, sqrt(counts).
2. Geoms (lines and ribbons vs points and intervals)
3. Scales (colouring and y-axis scaling)

For future maintainability, I think we could split this internally to 
- `.ppc_rootogram_data()` that handles the logic for data generation.
- define `geom_y <- ifelse ....` and `geom_yrep <- ifelse ....` and have one logical branch for constructing the final graph before returning it.